Fix highlighting in database debug toolbar

vlakoff opened the pull request:
Highlighting wasn't working for multiple-word terms, as they were searched in the SQL string with nbsp entities instead of regular spaces.

This code could be optimized, though it's not crucial as it's a debug function.

vlakoff commented:
Added a commit implementing replacement using regexp with word boundaries, to avoid overlaps / false positives with strings such as "OR", "IN"…

sfadschm commented:
Can you add before/after schreenshots?

paulbalandan commented on system/Database/Query.php:
```diff
@@ -406,10 +404,15 @@ public function debugToolbarDisplay(): string
+
+        $sql = str_replace(['(', ')'], ['<strong>(</strong>', '<strong>)</strong>'], $sql);
 
+        /** @noRector \Rector\CodeQuality\Rector\Return_\SimplifyUselessVariableRector */
```
I would agree with Rector here to simplify the return. Besides, our coding standard also includes this rule to simplify the return when possible (although it is not taking effect here).

vlakoff commented:
I confirmed the fix, but if ever you want screenshots of this toolbar content, you may have a look at #5136.

sfadschm commented:
> I confirmed the fix, but if ever you want screenshots of this toolbar content, you may have a look at #5136.

There is unfortunately none of the key words changed here visible in the other PR. Would be great to see just one example what has been fixed in this PR, too.

paulbalandan reviewed (changes requested):
Since this PR addresses an error in highlighting multiple-word identifiers, please add before/after screenshots, as @sfadschm suggested, that show there are indeed problems with the current and how this PR fixes them.

vlakoff commented:
Before:
![5129-before](https://user-images.githubusercontent.com/544424/135310611-d938b39d-67a5-459e-82fe-30e0affa6164.png)

After:
![5129-after](https://user-images.githubusercontent.com/544424/135310633-d9e135ce-2133-4b72-a603-c3252f0cab30.png)

vlakoff commented:
Maybe we should add the ASC and DESC keywords?

vlakoff commented:
Also, I'd suggest either highlighting commas in addition to parentheses, or not highlighting parentheses. Your pick?

sfadschm commented:
I think adding the keywords would be good, but would rather remove the parentheses. But lets wait for other calls.

kenjis commented:
I would like removing parentheses, adding ASC and DESC.

paulbalandan commented:
Let's highlight only keywords, leaving out parentheses and other punctuation marks.

vlakoff commented:
* Removed the highlighting of parentheses
* Added ASC and DESC keywords
* Reduced the number of commits by removing the commits related to the Rector warning

paulbalandan commented:
Thank you, @vlakoff
